### PR TITLE
codeintel: Simplify index enqueuer construction

### PR DESCRIPTION
**Pull request by efritz**
Make the index enqueuer an actual shared instance when constructing the frontend services. I was able to split the observability struct in the autoindex package between the structs that need them.

I also took the time to standardize creation and passing of observation contexts and operations for services.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/codeintel/background/indexing/iface.go [hunk 1]
```diff
@@ -20,3 +20,7 @@ type GitserverClient interface {
 	Head(ctx context.Context, repositoryID int) (string, error)
 	ListFiles(ctx context.Context, repositoryID int, commit string, pattern *regexp.Regexp) ([]string, error)
 }
+
+type IndexEnqueuer interface {
+	QueueIndex(ctx context.Context, repositoryID int) error
+}
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/background/indexing/iface.go:26]: nice, gets rid of having to make the mock enqueuer importable :sunglasses:

Comment (efritz) [reply enterprise/cmd/frontend/internal/codeintel/background/indexing/iface.go:26]: We can also split the tests so we don't need a live enqueuer (or mock downstream deps).

File: enterprise/internal/codeintel/autoindex/enqueuer/observability.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+type operations struct {
+	QueueIndex *observation.Operation
+}
+
+func newOperations(observationContext *observation.Context) *operations {
+	metrics := metrics.NewOperationMetrics(
+		observationContext.Registerer,
+		"codeintel_autoindex_enqueuer",
```
Comment (Strum355) [inline enterprise/internal/codeintel/autoindex/enqueuer/observability.go:17]: Does this need to be accounted for in /metrics stuff?

**Review by Strum355 — APPROVED** [review]

File: enterprise/internal/codeintel/autoindex/enqueuer/observability.go (hunk 2 above)
Comment (efritz) [reply enterprise/internal/codeintel/autoindex/enqueuer/observability.go:17]: Yeah, I lost the metrics after that change. Updated.

Thanks for the catch!

**Pull request merged**